### PR TITLE
Updated terminology.md

### PR DESCRIPTION
**Pull request by longsizhuo**
Key Additions:

Head and Tail of Data Structures:

New entries have been added to explicitly define the terms "head" and "tail" for data structures like linked lists, queues, and stacks. This enhancement is crucial for clarity, as these terms are fundamental in understanding and implementing these structures.
For instance, "头节点" (head node) and "尾节点" (tail node) for linked lists, "队首" (front of the queue) and "队尾" (rear of the queue) for queues, and "栈顶" (top of the stack) and "栈尾" (bottom of the stack) for stacks are now clearly mentioned.

### CODE REVIEW

File: docs-en/chapter_stack_and_queue/queue.md [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # Queue
 
-"Queue" is a linear data structure that follows the First-In-First-Out (FIFO) rule. As the name suggests, a queue simulates the phenomenon of lining up, where newcomers join the back of the queue, and people at the front of the queue leave one by one.
```
Comment (krahets) [inline docs-en/chapter_stack_and_queue/queue.md:3]: Please remove all the changes in `queue.md` in this PR

File: docs/chapter_appendix/terminology.md [hunk 2]
```diff
@@ -4,62 +4,69 @@
...
+| 列表       | list                           | 入度           | in-degree                   |
+| 动态数组     | dynamic array                  | 出度           | out-degree                  |
+| 硬盘       | hard disk                      | 邻接矩阵       | adjacency matrix            |
+| 内存       | RAM                            | 邻接表         | adjacency list              |
+| 缓存       | cache memory                   | 广度优先搜索   | breadth-first search        |
+| 缓存未命中    | cache miss                     | 深度优先搜索   | depth-first search          |
+| 缓存命中率    | cache hit rate                 | 二分查找       | binary search               |
+| 栈        | stack                          | 搜索算法       | searching algorithm         |
+| 栈顶       | top of the stack               | 排序算法       | sorting algorithm           |
+| 栈尾       | bottom of the stack            | 选择排序       | selection sort              |
```
Comment (krahets) [inline docs/chapter_appendix/terminology.md:38]: 栈底

File: docs/chapter_appendix/terminology.md [hunk 3]
```diff
@@ -4,62 +4,69 @@
...
+| 列表      | list                           | 入度      | in-degree                   |
+| 动态数组    | dynamic array                  | 出度      | out-degree                  |
+| 硬盘      | hard disk                      | 邻接矩阵    | adjacency matrix            |
+| 内存      | RAM                            | 邻接表     | adjacency list              |
+| 缓存      | cache memory                   | 广度优先搜索  | breadth-first search        |
+| 缓存未命中   | cache miss                     | 深度优先搜索  | depth-first search          |
+| 缓存命中率   | cache hit rate                 | 二分查找    | binary search               |
+| 栈       | stack                          | 搜索算法    | searching algorithm         |
+| 栈顶      | top of the stack               | 排序算法    | sorting algorithm           |
+| 栈尾      | bottom of the stack            | 选择排序    | selection sort              |
```
Comment (krahets) [inline docs/chapter_appendix/terminology.md:38]: 栈底

Comment (longsizhuo) [reply docs/chapter_appendix/terminology.md]: 抱歉，改好了

File: docs/chapter_appendix/terminology.md [hunk 4]
```diff
@@ -4,62 +4,69 @@
...
+| 算法      | algorithm                      | 层序遍历    | level-order traversal       |
+| 数据结构    | data structure                 | 广度优先遍历  | breadth-first traversal     |
+| 渐近复杂度分析 | asymptotic complexity analysis | 深度优先遍历  | depth-first traversal       |
+| 时间复杂度   | time complexity                | 二叉搜索树   | binary search tree          |
+| 空间复杂度   | space complexity               | 平衡二叉搜索树 | balanced binary search tree |
+| 迭代      | iteration                      | 平衡因子    | balance factor              |
+| 递归      | recursion                      | 堆       | heap                        |
+| 尾递归     | tail recursion                 | 大顶堆     | max heap                    |
+| 递归树     | recursion tree                 | 小顶堆     | min heap                    |
+| 大 O 记号  | big-O notation                 | 优先队列    | priority queue              |
```
Comment (krahets) [inline docs/chapter_appendix/terminology.md:18]: $O$

Please keep other glossaries invariant when adding the new glossaries.

**Review by krahets — APPROVED** [review]
Thanks for the PR!

**Pull request merged**